# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannelChild.cpp
# Commit: bfe21f274ed5
# Full Hash: bfe21f274ed5a4630a3503bcb875fed04f90533c
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-04-16 03:00:04
# Regressor Bug: 1594529
# File Overlap Count: 1
# Description:
#   Bug 1594529 - Create LoadInfo for subdocuments directly in parent process with DocumentChannel. r=mattwoodrow,nika
#   
#   Currently, with Fission enabled we are not able to create a proper LoadInfo
#   object when doing a subdocument load because we do not have access to a loading
#   context if the load is happening inside of an OOP frame. To solve this problem,
# ==============================================================================

diff -r 29ff609063c2 -r bfe21f274ed5 netwerk/ipc/DocumentChannelChild.cpp
--- a/netwerk/ipc/DocumentChannelChild.cpp	Wed Apr 15 18:53:22 2020 +0000
+++ b/netwerk/ipc/DocumentChannelChild.cpp	Wed Apr 15 18:53:06 2020 +0000
@@ -73,20 +73,19 @@
 
   gHttpHandler->OnOpeningDocumentRequest(this);
 
+  if (!GetDocShell() || !GetDocShell()->GetBrowsingContext() ||
+      GetDocShell()->GetBrowsingContext()->IsDiscarded()) {
+    return NS_ERROR_FAILURE;
+  }
+
   DocumentChannelCreationArgs args;
 
   args.loadState() = mLoadState->Serialize();
-  Maybe<LoadInfoArgs> maybeArgs;
-  rv = LoadInfoToLoadInfoArgs(mLoadInfo, &maybeArgs);
-  NS_ENSURE_SUCCESS(rv, rv);
-  MOZ_DIAGNOSTIC_ASSERT(maybeArgs);
-
-  args.loadInfo() = *maybeArgs;
   args.loadFlags() = mLoadFlags;
   args.cacheKey() = mCacheKey;
   args.channelId() = mChannelId;
   args.asyncOpenTime() = mAsyncOpenTime;
-  args.outerWindowId() = mLoadInfo->GetOuterWindowID();
+  args.outerWindowId() = GetDocShell()->GetOuterWindowID();
 
   Maybe<IPCClientInfo> ipcClientInfo;
   if (mInitialClientInfo.isSome()) {
@@ -107,11 +106,6 @@
     return NS_ERROR_ILLEGAL_VALUE;
   }
 
-  if (!GetDocShell() || !GetDocShell()->GetBrowsingContext() ||
-      GetDocShell()->GetBrowsingContext()->IsDiscarded()) {
-    return NS_ERROR_FAILURE;
-  }
-
   args.hasValidTransientUserAction() =
       GetDocShell()
           ->GetBrowsingContext()